src/main/kotlin/com/cycode/plugin/annotators/annotationAppliers/SastApplier.kt

```diff
@@ -0,0 +1,78 @@
+package com.cycode.plugin.annotators.annotationAppliers
+
+import com.cycode.plugin.CycodeBundle
+import com.cycode.plugin.annotators.convertSeverity
+import com.cycode.plugin.annotators.validateTextRange
+import com.cycode.plugin.cli.CliResult
+import com.cycode.plugin.cli.CliScanType
+import com.cycode.plugin.intentions.CycodeIgnoreIntentionQuickFix
+import com.cycode.plugin.intentions.CycodeIgnoreType
+import com.cycode.plugin.services.ScanResultsService
+import com.intellij.lang.annotation.AnnotationHolder
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.PsiFile
+
+class SastApplier(private val scanResults: ScanResultsService) : AnnotationApplierBase() {
+    private fun validateSastTextRange(textRange: TextRange, psiFile: PsiFile): Boolean {
+        // FIXME(MarshalX): for now, I dont see any way to validate the text range for SAST
+        //   small explanation:
+        //   - SAST doesn't provide end positions, so we have to calculate them from the line number (get the last character in the line)
+        //   - we can't use the same validation as for SCA because value in the range is unknown (for SCA we expect package name)
```

Comment:
Is it worth adding a Jira ticket to the SAST backlog for this, or is it already logged? Ideally it sounds like SAST should be returning start and end positions, but maybe this is not feasible (I don't know) 🤔

Reply:
i moved this Q to slack